apps/docs/src/guide/provider/index.md

```diff
@@ -0,0 +1,47 @@
+# Provider
+
+The [`Provider`](../../api/Account/Provider.md) class lets you connect to a Fuel node ([_*local*_](../introduction/getting-started.md#connecting-to-a-local-node) or [_*external*_](../introduction/getting-started.md#connecting-to-the-testnet)) and interact with it, encapsulating common client operations in the SDK.
+
+<<< @/../../docs-snippets/src/guide/provider/provider.test.ts#provider-definition{ts:line-numbers}
```

Comment:
After this section, I think it would be beneficial to have a section that gives people context about why the `Provider` is important, i.e. telling them where it is used in 'real-world' code snippets. 

For example, we shuold tell them that the provider is important because it is used every time you want to initialize a wallet, deploy a contract, read data from the chain, etc. All those use cases. We should give people a brief idea. Thoughts?

Reply:
You should probably also change the code snippet at the top and show something like:

- Initiating a provider
- Creating a wallet using the provider
- And then sending a transaction using a wallet

Reply:
@Dhaiwat10 I have updated the description. I also changed the code snippet, but opted for fetching the balance of the wallet instead of sending a transaction because the transaction first has to be prepared before being sent, which would take too much space and take focus away from the topic at hand which is the `Provider`.

I'll wait until #1961 is merged so that I can link to relevant pages so that the transaction sending part is referenced, but the text on the page seems okay to me right now.